appnav/src/main/kotlin/io/element/android/appnav/room/resolver/RoomAliasResolverView.kt

```diff
@@ -0,0 +1,205 @@
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import io.element.android.compound.theme.ElementTheme
+import io.element.android.libraries.architecture.AsyncData
+import io.element.android.libraries.designsystem.atomic.atoms.PlaceholderAtom
+import io.element.android.libraries.designsystem.atomic.pages.HeaderFooterPage
+import io.element.android.libraries.designsystem.components.avatar.AvatarSize
+import io.element.android.libraries.designsystem.components.button.BackButton
+import io.element.android.libraries.designsystem.preview.ElementPreview
+import io.element.android.libraries.designsystem.theme.components.Button
+import io.element.android.libraries.designsystem.theme.components.ButtonSize
+import io.element.android.libraries.designsystem.theme.components.CircularProgressIndicator
+import io.element.android.libraries.designsystem.theme.components.Text
+import io.element.android.libraries.designsystem.theme.components.TopAppBar
+import io.element.android.libraries.matrix.api.core.RoomId
+import io.element.android.libraries.ui.strings.CommonStrings
+
+@Composable
+fun RoomAliasResolverView(
+    state: RoomAliasResolverState,
+    onBackPressed: () -> Unit,
+    onAliasResolved: (RoomId) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val latestOnAliasResolved by rememberUpdatedState(onAliasResolved)
+    LaunchedEffect(state.resolveState) {
+        if (state.resolveState is AsyncData.Success) {
+            latestOnAliasResolved(state.resolveState.data)
+        }
+    }
+
+    HeaderFooterPage(
+        modifier = modifier,
+        paddingValues = PaddingValues(16.dp),
+        topBar = {
+            RoomAliasResolverTopBar(onBackClicked = onBackPressed)
+        },
+        content = {
+            RoomAliasResolverContent(state = state)
+        },
+        footer = {
+            RoomAliasResolverFooter(
+                state = state,
+            )
+        }
+    )
+}
+
+@Composable
+private fun RoomAliasResolverFooter(
+    state: RoomAliasResolverState,
+    modifier: Modifier = Modifier,
+) {
+    when (state.resolveState) {
+        is AsyncData.Failure -> {
+            Button(
+                text = stringResource(CommonStrings.action_retry),
+                onClick = {
+                    state.eventSink(RoomAliasResolverEvents.Retry)
+                },
+                modifier = modifier.fillMaxWidth(),
+                size = ButtonSize.Medium,
+            )
+        }
+        is AsyncData.Loading -> {
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.Center,
+            ) {
+                CircularProgressIndicator()
+            }
+        }
+        AsyncData.Uninitialized,
+        is AsyncData.Success -> Unit
+    }
+}
+
+@Composable
+private fun RoomAliasResolverContent(
+    state: RoomAliasResolverState,
+    modifier: Modifier = Modifier,
+) {
+    ContentScaffold(
```

Comment:
We should extract this in designsystem, add TODO?

Reply:
Done